lib/core/buffer.hpp

```diff
@@ -526,43 +528,58 @@ class Buffer : public GenericBuffer {
     /**
      * @brief Allocates a new frame description object holding a D dimensional
      *        array of type T
-     * @param[in] frame_id The frame ID of the frame to describe
      * @param[in] extents Array extentds in the D dimensions
      * @param[in] dimnames Array axis labels in the D dimensions
      */
     template<typename T, std::size_t D>
-    void allocate_new_frame_desc(int frame_id, kotekan::Symbol quantity_name,
+    void allocate_new_frame_desc(kotekan::Symbol quantity_name,
                                  const std::array<std::ptrdiff_t, D>& extents,
                                  const std::array<kotekan::Symbol, D>& dimnames) {
-        frames_desc.at(frame_id) =
-            std::make_shared<kotekan::NDArray<T, D>>(quantity_name, extents, dimnames, nullptr);
+        buffer_lock lock(mutex);
+        if (!frames_desc)
+            frames_desc =
+                std::make_shared<kotekan::NDArray<T, D>>(quantity_name, extents, dimnames, nullptr);
+        else {
+            assert(D == frames_desc->get_rank());
```

Comment:
Consider issuing kotekan errors; what behavior do we want in production here?

Reply:
done in 14298940a